### DISCUSSION TITLE
While hitting enter CP editor is not maintaining the same indent

### DISCUSSION THREAD

**Discussion by sahalhes** [cpeditor/cpeditor#1266] [Q&A]
I tried changing settings but don't know what i am missing. When I hit enter in normal code editor it puts cursor just below the line , but in CP editor the cursor is going to left automatically. Please tell what settings I should enable/disable ? 
![Screenshot (394)](https://github.com/cpeditor/cpeditor/assets/146409442/55108e89-73bd-43be-b990-6573efba7f29)
![Screenshot (393)](https://github.com/cpeditor/cpeditor/assets/146409442/d4654686-7324-4188-9c7e-a1cd1757775b)

**Comment by ouuan** [comment]
It should maintain the indent. I'm not sure what's wrong on your side. I can't see the text cursors clearly in your screenshot, and I'm not sure what space characters are in your code, and where the cursor was before pressing Enter.

**Reply by abhiiguptaa** [reply]
Please add the vim motions so that I can use this editor

**Comment by sahalhes** [comment]
> It should maintain the indent. I'm not sure what's wrong on your side. I can't see the text cursors clearly in your screenshot, and I'm not sure what space characters are in your code, and where the cursor was before pressing Enter.

Basically in line 6 four space was present, but by hitting enter to line 7 cursor is on the extreme left.

**Reply by ouuan** [reply]
Just want to make sure, where was the cursor before you hit enter?

**Comment by ouuan** [comment]
I'm not aware of any setting that leads to this wrong behavior. But you may try exporting the settings and then resetting to default (both in the Options menu) to see if that works.

**Reply by sahalhes** [reply]
Even in default settings it is showing the same way, don't know why
